src/python/pants/help/help_formatter.py

```diff
@@ -59,10 +61,10 @@ def add_option(ohis, *, category=None):
                 lines.extend(self.format_option(ohi))
 
         add_option(oshi.basic)
-        if self._show_recursive:
-            add_option(oshi.recursive, category="recursive")
```

Comment:
Good call to remove this. I agree that it's rare for us to have a recursive option and we don't want to emphasize it. Notably, we're removing the recursive options `--fmt-skip` and `--fmt-transitive`.